README.md

```diff
@@ -12,28 +12,185 @@ We believe that, like native platforms, the web deserves a first-class virtual s
+And, at least in v1, the following are out of scope:
 
-## Talk at Chrome Dev Summit 2018
+* Built-in support for advanced features and behaviors, such as those mentioned above.
+* Support infinite-grid (spreadsheet-like) or masonry layouts.
 
-At Chrome Dev Summit 2018, Gray Norton presented on our work so far. Give it a watch!
+## Proposed APIs
 
-[![virtual-scroller at Chrome Dev Summit 2018](http://img.youtube.com/vi/UtD41bn6kJ0/0.jpg)](http://www.youtube.com/watch?v=UtD41bn6kJ0)
+### `<virtual-content>` element
+
+The `<virtual-content>` element represents a container that will manage the rendering of its children.
+The children of this element might not get rendered/updated if they are not near or in the viewport.
+The element is aware of changes to the viewport, as well as to its own size, and will manage the rendered state of its children accordingly.
+
+Its children can be any element.
+Semantically, the element is similar to a `<div>`, with the addition of focus on only the visible contents when the element overflows its container and is scrolled.
+
+The sizes and other layout values of the non-rendered children, which might affect scrollbar height, etc., are approximate sizes and might not be always accurate.
+The rendered children always have accurate style and layout values, just like other normal DOM nodes.
+
+All children, rendered or non-rendered, will work with find-in-page, focus navigation, fragment URL navigation, and accessibility technology, just like normal DOM nodes.
+
+### `rangechange` event
+
+Fired when `<virtual-content>` is about to render a new range of items, e.g. because the user scrolled.
+This will fire at `requestAnimationFrame` timing, i.e. before the browser is about to paint.
+
+The event has the following properties:
+
+* `first`: an integer, the 0-based index of the first children currently rendered.
+* `last`: an integer, the 0-based index of the last children currently rendered.
+* `bubbles`: false
+* `cancelable`: false
+* `composed`: false
+
+As an example, this can be used to delay more costly rendering work.
+For example a scrolling code listing could perform just-in-time syntax highlighting on lines right before they become visible, leaving the un-adorned code accessible by find-in-page/etc. but improving the code's appearance before the user sees it.
+
+_TODO: do we need `first` and `last` on the event, or should we just use the properties of the element?_
+
+### `rangeFirst` and `rangeLast` getters
+
+These return 0-based indices giving the first and last children currently rendered.
+
+### Constraints
+
+Ideally we would like there to be zero constraints on the contents of the `<virtual-content>` element, or on the virtual-content element itself.
+
+This may prove difficult to specify or implement, so we may need to add constraints such as:
+
```

Comment:
Should we maybe also mention the "locked state"? Is that implementation detail?

Reply:
Yeah, that will be web-exposed, good point. I guess it's not much of a "constraint"... I'll see what I can do.